express-notes-api/index.js

```diff
@@ -0,0 +1,110 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+app.use(express.json());
+
+const data = require('./data.json');
+const notes = data.notes;
+
+app.get('/api/notes', (req, res) => {
+  if (req.params.id === undefined) {
+    res.send(notes);
+  }
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const parsedId = parseInt(req.params.id, 10);
+  if (notes[parsedId] !== undefined) {
+    res.send(notes[parsedId]);
+  } else if (isNaN(parsedId)) {
+    res.status(400).send({
+      error: 'id must be a positive integer'
+    });
+  } else {
+    res.status(400).send({
```

Comment:
if it's not found, we use `404`. Please double-check the instructions for the correct status codes.